### PR TITLE
RC 3.2.1

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Bug fix and maintenance release for pysat. Bug fixes required for pysatCDAAC and others.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

GitHub Actions

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs (include link [here](https://pysat.readthedocs.io/en/rc3_2_1/))
- [x] Upload the test distribution to the testPyPi [server](https://test.pypi.org/project/pysat/3.2.1rc3/)
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
  - [x] pysatCDAAC- [develop](https://github.com/pysat/pysatCDAAC/actions/runs/10617022389)
  - [x] pysatMadrigal - [develop](https://github.com/pysat/pysatMadrigal/actions/runs/10617130053)
  - [x] pysatMissions - [develop](https://github.com/pysat/pysatNASA/actions/runs/10616752888)
  - [ ] pysatModels - [enh/128_unzip](https://github.com/pysat/pysatModels/actions/runs/10617109823) -- delaying tests since multiple bugs need to be fixed here
  - [x] pysatNASA - [rc0.0.6](https://github.com/pysat/pysatNASA/actions/runs/10616752888)
  - [x] pysatSpaceWeather - [rc0.0.2](https://github.com/pysat/pysatSpaceWeather/actions/runs/10616669973)
- [x] Create a wiki page with the release text (put link [here](https://github.com/pysat/pysat/wiki/v3.2.1-Release-Text))
- [ ] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the
release checklist on the wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/tests/test_instruments.py [hunk 1]
```diff
@@ -77,7 +77,10 @@ def test_inst_start_time(self, inst_dict, kwarg, output):
 
         self.test_inst.load(date=date)
 
-        assert self.test_inst['uts'][0] == output
+        if self.test_inst.pandas_format:
+            assert self.test_inst['uts'].iloc[0] == output
+        else:
+            assert self.test_inst['uts'][0] == output
```
Comment (aburrell) [inline pysat/tests/test_instruments.py:83]: Would `assert self.test_inst[0, 'uts'] == output` work for this?

File: setup.cfg [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = pysat
-version = 3.2.0
+version = 3.2.1rc1
```
Comment (aburrell) [inline setup.cfg:3]: Don't forget to remove the 'rc' flag.

Comment (jklenzing) [inline setup.cfg:3]:
```suggestion
version = 3.2.1
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -12,12 +12,12 @@ jobs:
...
         os: ["ubuntu-latest", "windows-latest", "macos-latest"]
-        python-version: ["3.10", "3.11", "3.12"]
+        python-version: ["3.11", "3.12"]
         numpy_ver: ["latest"]
         test_config: ["latest"]
         include:
           # NEP29 compliance settings
-          - python-version: "3.9"
+          - python-version: "3.10"
             numpy_ver: "1.23"
```
Comment (jklenzing) [inline .github/workflows/main.yml:21]:
```suggestion
            numpy_ver: "1.24"
```

Comment (jklenzing) [reply .github/workflows/main.yml:21]: NEP29 for Jun 2024

File: pysat/tests/test_instruments.py (hunk 1 above)
Comment (jklenzing) [reply pysat/tests/test_instruments.py]: fixed

File: pysat/tests/test_instruments.py [hunk 4]
```diff
@@ -77,8 +77,7 @@ def test_inst_start_time(self, inst_dict, kwarg, output):
 
         self.test_inst.load(date=date)
 
-        assert self.test_inst['uts'][0] == output
-        return
+        assert self.test_inst[0, 'uts'] == output
```
Comment (aburrell) [inline pysat/tests/test_instruments.py:80]:
```suggestion
        assert self.test_inst[0, 'uts'] == output
        return
```

File: docs/examples/example_orbit.rst [hunk 5]
```diff
@@ -55,7 +55,7 @@ the pysat repository at ``demo/cnofs_vefi_dc_b_orbit_plots.py``.
        # pandas documentation for more information about the `resample` mtehod.
        # The 1-s cadence was chosen because it is the nominal cadence for this
        # instrument.
-       vefi.data = vefi.data.resample('1S', label='left').ffill(limit=1)
+       vefi.data = vefi.data.resample('1s', label='left').ffill(limit=1)
```
Comment (rstoneback) [inline docs/examples/example_orbit.rst:58]: Is this backwards compatible? On a different pull it looked like pandas doesn't have any overlap in the support for the keywords.

File: pysat/_constellation.py [hunk 6]
```diff
@@ -745,7 +745,7 @@ def to_inst(self, common_coord=True, fill_method=None):
             # Get the common coordinates needed for all data
             for cinst in self.instruments:
                 if not cinst.pandas_format:
-                    for new_coord in cinst.data.dims.keys():
+                    for new_coord in cinst.data.sizes.keys():
```
Comment (rstoneback) [inline pysat/_constellation.py:748]: How far back does .sizes work?

File: docs/examples/example_orbit.rst (hunk 5 above)
Comment (rstoneback) [reply docs/examples/example_orbit.rst:58]: Same comment for all of the frequency keyword changes.

Comment (jklenzing) [reply docs/examples/example_orbit.rst:58]: Currently passing tests in the operational branch for:
pandas-1.1.5

File: pysat/_constellation.py (hunk 6 above)
Comment (jklenzing) [reply pysat/_constellation.py:748]: Currently passing tests in the operational branch for:
xarray-0.16.2

File: docs/examples/example_orbit.rst (hunk 5 above)
Comment (jklenzing) [reply docs/examples/example_orbit.rst:58]: The overlap issue had to do with the year-end and month-end keywords, which were new with 2.2.

Comment (jklenzing) [reply docs/examples/example_orbit.rst:58]: Both the new keywords for 's' and 'ns' work as expected in pandas 1.5.0.

**Comment by jklenzing** [comment]
@rstoneback, I have added #1208 to capture the minimum dependency concerns. Implementation can be straightforward, but actually determining what these minimum versions is not. I recommend we discuss that more fully for the next release.

**Review by JonathonMSmith — APPROVED** [review]

**Review by rstoneback — APPROVED** [review]

**Pull request merged**